[{Teo Mou Qi}] iP

kimberlytmq opened the pull request:
### Cloudy

> "Your mind is for having ideas, not holding them." - David Allen [source](https://dansilvestre.com/productivity-quotes)

Cloudy frees your mind of having to remember things you need to do. It's

- **text-based**
- **easy** to learn
- **fast** to use

All you need to do is,

1. Download it from [here](https://github.com/kimberlytmq/ip/releases/tag/A-Jar)
2. Double-click it
3. Add your tasks
4. Let it manage tasks for you!😉

Best of all, it's **_FREE_**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:
`public static void main(String[] args) {
        new Cloudy("data/tasks.txt").run();
    }`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kimberlytmq committed:
Level-0 task

kimberlytmq committed:
Level-1 task

kimberlytmq committed:
Level 2 task

kimberlytmq committed:
Level 3 task

kimberlytmq committed:
level 4 tasks

kimberlytmq committed:
Level 5 task

kimberlytmq committed:
Level 6 task

kimberlytmq committed:
TextUiTesting

kimberlytmq committed:
Add txt file to save data

kimberlytmq committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add txt file to save data

kimberlytmq committed:
Add date feature for deadlines and events

kimberlytmq committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add date feature for deadlines and events

kimberlytmq committed:
Update code to OOP

kimberlytmq committed:
Add files to package cloudy

kimberlytmq committed:
Merge remote-tracking branch 'origin/add-gradle-support'

* origin/add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

kimberlytmq committed:
Add project package to build.gradle

kimberlytmq committed:
Add Junit tests

kimberlytmq committed:
Add Javadoc comments

kimberlytmq committed:
Align with coding standard

kimberlytmq committed:
Add find feature

kimberlytmq committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Align with coding standard

kimberlytmq committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add find feature

# Conflicts:
#	src/main/java/cloudy/Cloudy.java
#	src/main/java/cloudy/Ui.java

JoanneHing commented on src/main/java/cloudy/Cloudy.java:
```diff
@@ -0,0 +1,185 @@
+                    break;
+                case "find":
+                    handleFindCommand(command);
+                    break;
+                case "mark":
+                    handleMarkCommand(command);
+                    break;
+                case "unmark":
+                    handleUnmarkCommand(command);
+                    break;
+                case "todo":
+                    handleTodoCommand(command);
+                    break;
+                case "deadline":
+                    handleDeadlineCommand(command);
+                    break;
+                case "event":
+                    handleEventCommand(command);
+                    break;
+                case "delete":
+                    handleDeleteCommand(command);
+                    break;
+                case "invalid":
+                    ui.showInvalidCommand();
+                    break;
+                case "invalidTaskNum":
+                    ui.showInvalidTaskNum();
+                    break;
+                case "invalidTaskFormat":
+                    ui.showInvalidTaskFormat();
+                    break;
+                case "invalidDeadline":
+                    ui.showInvalidDeadline();
+                    break;
+                case "invalidEvent":
+                    ui.showInvalidEvent();
+                    break;
+                default:
+                    ui.showInvalidCommand();
+            }
```
There should be no indentation for case clauses in switch statements, maybe can configure your IDE to follow the styling

Can refer to [here](https://se-education.org/guides/conventions/java/intermediate.html#conditionals) under the Layout section

JoanneHing commented on src/main/java/cloudy/Command.java:
Overall class is neat! But the public methods is missing the header comment file for the JavaDoc, may add in for it.

JoanneHing commented on src/main/java/cloudy/Parser.java:
```diff
@@ -0,0 +1,191 @@
+        String searchQuery = userInput.substring(5).trim();
+        return new Command("find", searchQuery);
+    }
+
```
Oops, the header comment for this task is missing too

JoanneHing commented on src/main/java/cloudy/Task.java:
```diff
@@ -0,0 +1,40 @@
+        this.description = description;
+        this.isMarked = isMarked;
+    }
+
+    public void markTask() {
+        this.isMarked = true;
+    }
+
+    public void unmarkTask() {
+        this.isMarked = false;
+    }
+
+    public boolean isMarked() {
+        return isMarked;
+    }
```
Header comment is missing for these methods too, may update on it!

JoanneHing commented on src/main/java/cloudy/TaskList.java:
```diff
@@ -0,0 +1,65 @@
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = new ArrayList<>(tasks);
+    }
+
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    public Task getTask(int index) {
+        if (index >= 0 && index < tasks.size()) {
+            return tasks.get(index);
+        }
+        return null;
+    }
+    public void removeTask(int index) {
+        if (index >= 0 && index < tasks.size()) {
+            tasks.remove(index);
+        }
+    }
```
Missing header comment and a space line could be added between line 40 and 41

JoanneHing commented on src/main/java/cloudy/Ui.java:
```diff
@@ -0,0 +1,122 @@
+package cloudy;
+
+<<<<<<< HEAD
+/**
+ * The Ui class handles all interactions with the user.
+ * It is responsible for displaying messages, including greetings, task lists,
+ * and error messages, in a formatted manner.
+ */
+=======
+import java.util.ArrayList;
+
+>>>>>>> branch-Level-9
```
Ops, I think this comment on merge conflict is missed out from being removed

JoanneHing reviewed (commented):
Overall LGTM! Just that some of the JavaDoc for the public methods is missed out. And the styling for switch statement need to be fix for the indentation of the `case` clauses. Can review for the comment under `Cloudy.java`. Good job!

kimberlytmq committed:
Edit comment

adriantan101 commented on src/main/java/cloudy/Cloudy.java:
```diff
@@ -0,0 +1,185 @@
+            userInput = echo.nextLine();
+            Command command = parser.parseCommand(userInput);
+
+            switch (command.getType()) {
+                case "bye":
+                    ui.showBye();
+                    return;
+                case "list":
+                    ui.showList(tasks);
+                    break;
+                case "find":
+                    handleFindCommand(command);
+                    break;
+                case "mark":
+                    handleMarkCommand(command);
+                    break;
+                case "unmark":
+                    handleUnmarkCommand(command);
+                    break;
+                case "todo":
+                    handleTodoCommand(command);
+                    break;
+                case "deadline":
```
I like how naming conventions are always adhered to 👍 package, variable and method names have consistent styling!

adriantan101 commented on src/main/java/cloudy/Cloudy.java:
```diff
@@ -0,0 +1,185 @@
+                case "deadline":
+                    handleDeadlineCommand(command);
+                    break;
+                case "event":
+                    handleEventCommand(command);
+                    break;
+                case "delete":
+                    handleDeleteCommand(command);
+                    break;
+                case "invalid":
+                    ui.showInvalidCommand();
+                    break;
+                case "invalidTaskNum":
+                    ui.showInvalidTaskNum();
+                    break;
+                case "invalidTaskFormat":
+                    ui.showInvalidTaskFormat();
+                    break;
+                case "invalidDeadline":
+                    ui.showInvalidDeadline();
+                    break;
+                case "invalidEvent":
```
Can consider changing IntelliJ settings to make switch statements have proper indentations

adriantan101 commented on src/main/java/cloudy/Ui.java:
```diff
@@ -0,0 +1,121 @@
+package cloudy;
+
+
+/**
+ * The Ui class handles all interactions with the user.
+ * It is responsible for displaying messages, including greetings, task lists,
+ * and error messages, in a formatted manner.
+ */
+
+import java.util.ArrayList;
+
+public class Ui {
+
+    /**
+     * Displays the greeting message when the program starts.
+     */
```
Good job for the JavaDoc comments written 👍

adriantan101 commented on src/main/java/cloudy/Parser.java:
```diff
@@ -0,0 +1,191 @@
+        String[] parts = userInput.split(" ");
+        if (parts.length == 2) {
+            try {
+                int taskNumber = Integer.parseInt(parts[1]);
+                return new Command("unmark", taskNumber);
```
Can consider adding more error messages/logging in the catch blocks to help with debugging 🫡

adriantan101 commented on src/main/java/cloudy/Parser.java:
```diff
@@ -0,0 +1,191 @@
+     * @param userInput The user's input containing the mark command.
+     * @return A command object for marking a task as complete.
+     */
+    private Command parseMarkCommand(String userInput) {
+        String[] parts = userInput.split(" ");
+        if (parts.length >= 2) {
+            try {
+                int taskNumber = Integer.parseInt(parts[1]);
+                return new Command("mark", taskNumber);
+            } catch (NumberFormatException e) {
+                return new Command("invalidTaskNum");
+            }
+        } else {
+            return new Command("invalidTaskFormat");
+        }
+    }
```
There are repeated patterns in methods like parseMarkCommand, parseUnmarkCommand, and parseDeleteCommand. Can consider abstracting this into a helper method 👍

adriantan101 reviewed (commented):
Other than the style issue with the switch statements, overall naming conventions and coding standards are adhered to! LGTM 👍

JoanneHing commented on src/main/java/cloudy/Cloudy.java:
I like how the naming for the method is named well so it is easy to understand how the methods works! well done!

JoanneHing commented on src/main/java/cloudy/Cloudy.java:
I like how the naming for the method is named well so it is easy to understand how the methods works! well done!

JoanneHing commented on src/main/java/cloudy/Command.java:
```diff
@@ -0,0 +1,67 @@
+ * Provides methods to initialise a new instance of a command and to get task information
+ */
+public class Command {
+    private String type;
+    private int taskNumber;
+    private String taskDescription;
+    private LocalDate deadline;
+    private LocalDate startTime;
+    private LocalDate endTime;
+
```
Variables are named well to understand their representative too! well done1

JoanneHing commented on src/main/java/cloudy/Parser.java:
```diff
@@ -0,0 +1,191 @@
+    }
+
+    private Command parseDeadlineCommand(String userInput) {
+        String[] parts = userInput.split("/by ");
```
maybe `parts` can be replaced by a more meaningful name? For example `SplitedInputs`

JoanneHing commented on src/main/java/cloudy/Parser.java:
```diff
@@ -0,0 +1,191 @@
+        if (parts.length < 3) {
+            return null;
+        }
+        String typeOfTask = parts[0];
+        boolean isMarked = parts[1].equals("1");
+        String taskDescription = parts[2];
```
good naming convention for the value represented by the splitted parts!

JoanneHing reviewed (commented):
Overall the code quality is maintained well, the naming for the class, variables and methods is done well! Good job1

kimberlytmq committed:
Add GUI

kimberlytmq committed:
Add assertions

The code seems to work fine, but there can be more ways to check for errors in the code.

Assertions can define assumptions about the program state so that the runtime can verify them.

kimberlytmq committed:
Improve code quality

Avoid long methods, change deep nesting to avoid arrowhead style code, and structure the code logically

kimberlytmq committed:
Add streams

Streams can make the code cleaner and more expressive. InStream allows for indexed iteration, so it is easy to replace loops when dealing with collections like TaskList.

The showList and handleFindCommand methods now return the input using streams.

kimberlytmq committed:
Merge pull request #1 from kimberlytmq/branch-A-Assertions

Add assertions

kimberlytmq committed:
Merge branch 'master' into branch-A-CodeQuality

kimberlytmq committed:
Merge pull request #2 from kimberlytmq/branch-A-CodeQuality

Improve code quality

kimberlytmq committed:
Merge branch 'master' into branch-A-Streams

kimberlytmq committed:
Merge pull request #3 from kimberlytmq/branch-A-Streams

Add streams

kimberlytmq committed:
Merge branch 'master' of https://github.com/kimberlytmq/ip

* 'master' of https://github.com/kimberlytmq/ip:
  Add streams
  Improve code quality
  Add assertions

kimberlytmq committed:
Test app and ensure that abstract class 'Task' works

kimberlytmq committed:
Revert "Test app and ensure that abstract class 'Task' works"

This reverts commit 680c40cc80086093b1a36748cb2c3567d60211a3.

kimberlytmq committed:
Add comments for methods in abstract class Task

kimberlytmq committed:
Change GUI

kimberlytmq committed:
Change GUI background colour

kimberlytmq committed:
Improve GUI text bubbles

kimberlytmq committed:
Add Ui screenshot

kimberlytmq committed:
Update user guide in README

kimberlytmq committed:
Add image to README

kimberlytmq committed:
Change gradle name for build

kimberlytmq committed:
Fix find command to have custom error messages
